compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -781,6 +787,51 @@ private String calculateResourceFunctionName(IdentifierToken accessorName, NodeL
         return sb.toString();
     }
 
+    private void createAnonymousTypeDefForConstantDeclaration(BLangConstant constantNode, Location pos,
+                                                               Location identifierPos) {
+        NodeKind nodeKind = constantNode.expr.getKind();
+
+        // Create a new literal.
+        BLangLiteral literal;
+        if (nodeKind == NodeKind.LITERAL) {
+            literal = (BLangLiteral) TreeBuilder.createLiteralExpression();
+        } else {
+            literal = (BLangLiteral) TreeBuilder.createNumericLiteralExpression();
+        }
+        if (nodeKind == NodeKind.LITERAL || nodeKind == NodeKind.NUMERIC_LITERAL) {
+            literal.setValue(((BLangLiteral) constantNode.expr).value);
+            literal.setOriginalValue(((BLangLiteral) constantNode.expr).originalValue);
+            literal.setBType(constantNode.expr.getBType());
+        } else {
+            // Since we only allow literals and unary expressions to come to this point we can straightaway
```

Comment:
Only unary expressions?

Reply:
yes, the function that leads here, allows only literals to be assigned to constants, however, since we special cased that to allow unary as well, after checking for literals we are only left with unary  expressions

Reply:
Meant that the comment wasn't quite right.